src/deposit/execute_deposit_utils.cairo

```diff
@@ -80,29 +104,392 @@ struct ExecuteDepositCache {
+        // it is possible that the addition of the positive impact amount of tokens into the pool
+        // could increase the imbalance of the pool, for most cases this should not be a significant
+        // change compared to the improvement of balance from the actual deposit
+        let positive_impact_amount = to_unsigned(
+            market_utils::apply_swap_impact_with_cap(
+                *params.data_store,
+                *params.event_emitter,
+                *_params.market.market_token,
+                *_params.token_out,
+                *_params.token_out_price,
+                to_signed(price_impact_usd, true),
+            )
+        );
+
+        // calculate the usd amount using positiveImpactAmount since it may
+        // be capped by the max available amount in the impact pool
+        // use tokenOutPrice.max to get the USD value since the positiveImpactAmount
+        // was calculated using a USD value divided by tokenOutPrice.max
+        //
+        // for the initial deposit, the pool value and token supply would be zero
+        // so the market token price is treated as 1 USD
+        //
+        // it is possible for the pool value to be more than zero and the token supply
+        // to be zero, in that case, the market token price is also treated as 1 USD
+        mint_amount =
+            market_utils::usd_to_market_token_amount(
+                positive_impact_amount * *_params.token_out_price.max,
+                to_unsigned(pool_value),
+                market_tokens_supply,
+            );
+
+        market_utils::apply_delta_to_pool_amount(
+            *params.data_store,
+            *params.event_emitter,
+            *_params.market,
+            *_params.token_out,
+            to_signed(positive_impact_amount, false),
+        );
+
+        market_utils::validate_pool_amount(params.data_store, _params.market, *_params.token_out,);
+
+        if (price_impact_usd < 0) {
+            // when there is a negative price impact factor,
+            // less of the deposit amount is used to mint market tokens
+            // for example, if 10 ETH is deposited and there is a negative price impact
+            // only 9.995 ETH may be used to mint market tokens
+            // the remaining 0.005 ETH will be stored in the swap impact pool
+            let negative_impact_amount = market_utils::apply_swap_impact_with_cap(
+                *params.data_store,
+                *params.event_emitter,
+                *_params.market.market_token,
+                *_params.token_out,
+                *_params.token_out_price,
+                to_signed(price_impact_usd, false),
+            );
+
+            amount_after_fees -= to_unsigned((-negative_impact_amount));
+        }
+    }
+
+    mint_amount +=
+        market_utils::usd_to_market_token_amount(
+            amount_after_fees * *_params.token_in_price.min,
+            to_unsigned(pool_value),
+            market_tokens_supply,
+        );
+
```

Comment:
Missing call to MarketUtils.applyDeltaToPoolAmount and MarketUtils.validatePoolAmount